openedx/core/djangoapps/catalog/utils.py

```diff
@@ -101,18 +106,63 @@ def get_program_types(user=None):  # pylint: disable=redefined-builtin
         return []
 
 
-def get_programs_data(user=None):
-    """Return the list of Programs after adding the ProgramType Logo Image"""
-
-    programs_list = get_programs(user)
-    program_types = get_program_types(user)
-
-    program_types_lookup_dict = {program_type["name"]: program_type for program_type in program_types}
+def _get_program_instructors(program):
+    """
+    Returns the list of instructor from cached if cache key exists otherwise
+    iterate over the courses and return all the instructors of each course run
+    """
+    cache_key = 'program.instructors.{program_id}'.format(
+        program_id=program.get('uuid')
+    )
+
+    lookup_list = []
+    instructors = cache.get(cache_key) or []
+    if instructors:
+        return instructors
+
+    for queryset in CourseOverview.objects.filter(id__in=_get_all_course_run_keys(program)).values_list(
+            'instructor_info', flat=True):
+        queryset = json.loads(queryset)
+        for instructor in queryset.get("instructors", []):
+            if instructor.get('name') not in lookup_list:
+                lookup_list.append(instructor.get('name'))
+                instructors.append(instructor)
+    cache.set(cache_key, instructors, app_settings.CACHE_TOOLBOX_DEFAULT_TIMEOUT)
+    return instructors
+
+
+def _get_all_course_run_keys(program):
+    """
+    Returns the course keys of all the course runs of a program.
+    """
+    keys = []
+    for course in program.get("courses", []):       # pylint: disable=E1101
+        for course_run in course.get("course_runs", []):
+            keys.append(CourseKey.from_string(course_run.get("key")))
+    return keys
 
-    for program in programs_list:
-        program["logo_image"] = program_types_lookup_dict[program["type"]]["logo_image"]
 
-    return programs_list
+def get_active_programs_data(user=None, program_id=None):
```

Comment:
I think we should have two separate functions. One for getting the list of active programs and one for getting the single program since we are doing something different in each case (instructor info vs. no instructor info).

The single program function should return the program data including program type and instructor info regardless of status. We can check status == active in the program detail view.